src/geometry/WorkingGeometry.ts

```diff
@@ -110,6 +110,83 @@ export class WorkingGeometry {
         };
     }
 
+    /**
+     * Translate
+     *
+     * @param {vec3} v: translation vector
+     */
+    public translate(v: vec3) {
+        const translationMatrix: mat4 = mat4.fromTranslation(mat4.create(), v);
+        this.transform(translationMatrix);
+    }
+
+    /**
+     * Rotate
+     *
+     * @param {vec3} axis: axis of rotation
+     * @param {number} angle: amount to rotate in radians
+     * @param {vec3} holdPoint: point to rotate from, default to true origin
+     */
+    public rotate(axis: vec3, angle: number, holdPoint: vec3 = vec3.create()) {
+        const normalAxis = vec3.normalize(vec3.create(), axis);
+        const quartnion = quat.setAxisAngle(quat.create(), normalAxis, angle);
+        const rotationMatrix: mat4 = mat4.fromRotationTranslationScaleOrigin(
+            mat4.create(),
+            quartnion,
```

Comment:
small thing: should this be "quaternion"?

Reply:
Yes this should be `quaternion`

Reply:
Done